cborg/src/Codec/CBOR/Decoding.hs

```diff
@@ -131,13 +136,15 @@ data DecodeAction s a
     | ConsumeTag64     (Word64# -> ST s (DecodeAction s a))
 #endif
 
-    | ConsumeInteger (Integer    -> ST s (DecodeAction s a))
-    | ConsumeFloat   (Float#     -> ST s (DecodeAction s a))
-    | ConsumeDouble  (Double#    -> ST s (DecodeAction s a))
-    | ConsumeBytes   (ByteString -> ST s (DecodeAction s a))
-    | ConsumeString  (Text       -> ST s (DecodeAction s a))
-    | ConsumeBool    (Bool       -> ST s (DecodeAction s a))
-    | ConsumeSimple  (Word#      -> ST s (DecodeAction s a))
+    | ConsumeInteger       (Integer   -> ST s (DecodeAction s a))
+    | ConsumeFloat         (Float#    -> ST s (DecodeAction s a))
+    | ConsumeDouble        (Double#   -> ST s (DecodeAction s a))
+    | ConsumeBytes         (ByteString-> ST s (DecodeAction s a))
+    | ConsumeByteArray     (BA.ByteArray -> ST s (DecodeAction s a))
+    | ConsumeString        (Text      -> ST s (DecodeAction s a))
+    | ConsumeUtf8ByteArray (BA.ByteArray -> ST s (DecodeAction s a))
```

Comment:
It's not ambiguous but it may be confusing. The problem is that we have to worry about two `ByteArray`s: `Data.Primitive.ByteArray.ByteArray` and `Codec.CBOR.ByteArray.ByteArray`. I found it easier to reason about if I was always explicit about which I was talking about.